packages/smart-wallet/src/smartWallet.js

```diff
@@ -376,27 +519,34 @@ export const prepareSmartWallet = (baggage, shared) => {
          *
          * If the purse doesn't exist, we hold the payment in durable storage.
          *
-         * @param {import('@endo/far').FarRef<Payment>} payment
+         * @param {Payment} payment
          * @returns {Promise<Amount>} amounts for deferred deposits will be empty
          */
         async receive(payment) {
+          const { helper } = this.facets;
           const { paymentQueues: queues, bank, invitationPurse } = this.state;
           const { registry, invitationBrand } = shared;
           const brand = await E(payment).getAllegedBrand();
 
           // When there is a purse deposit into it
           if (registry.has(brand)) {
             const purse = E(bank).getPurse(brand);
-            // @ts-expect-error deposit does take a FarRef<Payment>
             return E(purse).deposit(payment);
           } else if (invitationBrand === brand) {
-            // @ts-expect-error deposit does take a FarRef<Payment>
+            // @ts-expect-error narrow assetKind to 'set'
             return E(invitationPurse).deposit(payment);
           }
 
+          const purse = await helper.providePurseForKnownBrand(brand);
+          if (purse) {
+            return E(purse).deposit(payment);
+          }
+
           // When there is no purse, save the payment into a queue.
           // It's not yet ever read but a future version of the contract can
+          console.warn(`cannot deposit payment with brand ${brand}: no purse`);
           appendToStoredArray(queues, brand, payment);
+          // @@NOTE: default 'nat' assetKind is not always right here
```

Comment:
is this tech debt? what are the implications of `0n` for the `value` of a non-nat Amount?

Reply:
oops... `@@` meant that I intended to deal with it before marking it rfr.

I wonder if we should throw here. Since we didn't deposit it into a purse, we don't actually have exclusive access to the payment. We're in a race with the caller to see who spends it first.

Reply:
from discussion: agree it should throw instead of empty. In general `receive` can fail. It does fail in this moment, even though it saves the payment to try again later.